core/src/main/java/com/novoda/merlin/service/Ping.java

```diff
@@ -14,12 +15,16 @@
 
     public boolean doSynchronousPing() {
         MerlinLog.d("Pinging : " + hostAddress);
-        int responseCode = responseCodeFetcher.from(hostAddress);
-        MerlinLog.d("Got response : " + responseCode);
-        return isSuccess(responseCode);
-    }
+        try {
+            responseCodeFetcher.from(hostAddress);
+        } catch (RequestException e) {
+            if (e.causedByIO()) {
```

Comment:
I agree about no longer allowing a client defined url but this PR just broke the real connectivity check eg being connected to a guest wifi hotspot should not return success❌